version: "2"
services:
  giftless:
    image: datopian/giftless:0.3.0
    environment:
      - GIFTLESS_CONFIG_FILE=giftless.yaml
    user: "60421:60421"
    command:
      - "--strict"
      # we change 127.0.0.1 to 0.0.0.0 so the port is accessible outside the container
      - "--uwsgi-socket"
      - "0.0.0.0:5000"
      # the stats port is not currently exposed outside the container
      - "--stats"
      - "127.0.0.1:9191"
      - "--master"
      - "--enable-threads"
      - "--threads"
      - "8"
      - "--processes"
      - "2"
      - "--post-buffering"
      - "1"
      - "--socket-timeout"
      - "1200"
      - "--die-on-term"
      - "--manage-script-name"
      - "--callable"
      - "app"
    restart: always
    volumes:
      # Config file
      - "./giftless.yaml:/app/giftless.yaml"
      # Our auth plugin
      - "./token_database.py:/app/giftless/auth/token_database.py"
      # Database used by auth plugin
      - "/var/lib/git-lfs-authenticate:/var/lib/git-lfs-authenticate"
      # Where actual LFS file contents are stored
      - "/data/lfs/storage:/lfs-storage"
    ports:
      - "6421:5000"
